### PR TITLE
Sync to CI01133

### PR DESCRIPTION
**Pull request by superboyiii**

### CODE REVIEW

File: neo-cli/CLI/MainService.Contracts.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ private void OnDeployCommand(string filePath, string manifestPath = null)
         /// <param name="sender">Transaction's sender</param>
         /// <param name="signerAccounts">Signer's accounts</param>
         [ConsoleCommand("invoke", Category = "Contract Commands")]
-        private void OnInvokeCommand(UInt160 scriptHash, string operation, JArray contractParameters = null, UInt160 sender = null, UInt160[] signerAccounts = null)
+        private void OnInvokeCommand(UInt160 scriptHash, string operation, bool hasReturnValue = true, JArray contractParameters = null, UInt160 sender = null, UInt160[] signerAccounts = null)
```
Comment (shargon) [inline neo-cli/CLI/MainService.Contracts.cs:51]: Can we get this value from the manifest?

Comment (superboyiii) [reply neo-cli/CLI/MainService.Contracts.cs]: > Can we get this value from the manifest?

Done.

File: neo-cli/CLI/MainService.cs [hunk 2]
```diff
@@ -525,11 +525,22 @@ private bool OnInvokeWithResult(UInt160 scriptHash, string operation, out StackI
...
+            var snapshot = Blockchain.Singleton.GetSnapshot();
+            ContractState contract = NativeContract.ContractManagement.GetContract(snapshot, scriptHash);
+            if (contract == null)
+            {
+                Console.WriteLine("Contract is not exit");
+                result = StackItem.Null;
+                return false;
+            }
+
+            bool hasReturnValue = !contract.Manifest.Abi.GetMethod(operation).ReturnType.ToString().Equals("Void");
```
Comment (shargon) [inline neo-cli/CLI/MainService.cs:537]: We should check if the method exists

**Comment by superboyiii** [comment]
Test **PASS**

```
neo> balanceOf 0x0e1b9bfaa44e60311f6f3c96cfcd6d12c2fc3add NLezY74oBNBQqDgCu2tvwjwgmx8VV8ZuvD
Invoking script with: 'DBQIIwOU9053zvDVBdYLP1B9aycCFxERHwwJYmFsYW5jZU9mDBTdOvzCEm3Nz5Y8bx8xYE6k+psbDkFifVtS'
VM State: HALT
Gas Consumed: 0.0198409
Result Stack: [{"type":"Integer","value":"100000000"}]

NeoToken balance: 100000000
neo> open wallet cn.json
password: *
neo> register candidate NLGXdtqtQbSqxMnMD3FDudkCbA5e21L94n
Invoking script with: 'DCEDBUiiYOzZfo03x1f7m6ayvEcA/kUwoQKA6zew2dCCs+sRER8MEXJlZ2lzdGVyQ2FuZGlkYXRlDBTdOvzCEm3Nz5Y8bx8xYE6k+psbDkFifVtS'
VM State: HALT
Gas Consumed: 0.0598409
Result Stack: [{"type":"Boolean","value":true}]
Relay tx(no|yes): y
Signed and relayed transaction with hash=0x85c525c115c7fbe5e61731968d12f5699f0c9f84dd11081d13f3127d1443eb05
neo> open wallet admin.json
password: *
neo> get candidates
Invoking script with: 'EBEfDA1nZXRDYW5kaWRhdGVzDBTdOvzCEm3Nz5Y8bx8xYE6k+psbDkFifVtS'
VM State: HALT
Gas Consumed: 1.0098385

Candidates:
030548a260ecd97e8d37c757fb9ba6b2bc4700fe4530a10280eb37b0d9d082b3eb      0
neo> vote NLezY74oBNBQqDgCu2tvwjwgmx8VV8ZuvD 030548a260ecd97e8d37c757fb9ba6b2bc4700fe4530a10280eb37b0d9d082b3eb
Invoking script with: 'DCEDBUiiYOzZfo03x1f7m6ayvEcA/kUwoQKA6zew2dCCs+sMFAgjA5T3TnfO8NUF1gs/UH1rJwIXEhEfDAR2b3RlDBTdOvzCEm3Nz5Y8bx8xYE6k+psbDkFifVtS'
VM State: HALT
Gas Consumed: 0.0598433
Result Stack: [{"type":"Boolean","value":true}]
Relay tx(no|yes): y
Signed and relayed transaction with hash=0x2ed1de660c90788f95efa379935dbd988c3429101056ddfa9aca1551593e81d0
neo> get committee
Invoking script with: 'EBEfDAxnZXRDb21taXR0ZWUMFN06/MISbc3PljxvHzFgTqT6mxsOQWJ9W1I='
VM State: HALT
Gas Consumed: 1.0098385

Committee:
030548a260ecd97e8d37c757fb9ba6b2bc4700fe4530a10280eb37b0d9d082b3eb
neo> get candidates
Invoking script with: 'EBEfDA1nZXRDYW5kaWRhdGVzDBTdOvzCEm3Nz5Y8bx8xYE6k+psbDkFifVtS'
VM State: HALT
Gas Consumed: 1.0098385

Candidates:
030548a260ecd97e8d37c757fb9ba6b2bc4700fe4530a10280eb37b0d9d082b3eb      100000000
neo> invoke 0x0e1b9bfaa44e60311f6f3c96cfcd6d12c2fc3add decimals
Invoking script with: 'EBEfDAhkZWNpbWFscwwU3Tr8whJtzc+WPG8fMWBOpPqbGw5BYn1bUg=='
VM State: HALT
Gas Consumed: 0.0098385
Result Stack: [{"type":"Integer","value":"0"}]
Relay tx(no|yes): n
neo> invoke 0x0e1b9bfaa44e60311f6f3c96cfcd6d12c2fc3ade decimals
Contract does not exist.
neo> invoke 0x0e1b9bfaa44e60311f6f3c96cfcd6d12c2fc3ade decimal
Contract does not exist.
neo> invoke 0x0e1b9bfaa44e60311f6f3c96cfcd6d12c2fc3add decimal
This method does not not exist in this contract.
```

File: neo-cli/CLI/MainService.cs [hunk 3]
```diff
@@ -525,11 +525,34 @@ private bool OnInvokeWithResult(UInt160 scriptHash, string operation, out StackI
...
+            {
+                if (contract.Manifest.Abi.GetMethod(operation) == null)
+                {
+                    Console.WriteLine("This method does not not exist in this contract.");
+                    result = StackItem.Null;
+                    return false;
+                }
+                else
+                {
+                    hasReturnValue = !contract.Manifest.Abi.GetMethod(operation).ReturnType.ToString().Equals("Void");
```
Comment (shargon) [inline neo-cli/CLI/MainService.cs:547]:
```suggestion
                    hasReturnValue = contract.Manifest.Abi.GetMethod(operation).ReturnType != ContractParameterType.Void;
```

**Comment by superboyiii** [comment]
Done

**Review by shargon — APPROVED** [review]

**Pull request merged**